src/actions/check-solution/index.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import * as core from '@actions/core';
+import * as artifact from '@actions/artifact';
+import { ActionLogger, AuthHandler, AuthKind, getWorkingDirectory, PacRunner } from '../../lib';
+import path = require('path');
+import fs = require('fs-extra');
+
+core.startGroup('check-solution:');
+const workingDir = getWorkingDirectory('working-directory', false);
+const solutionPathCandidate = core.getInput('path', { required: true });
+const solutionPath = path.isAbsolute(solutionPathCandidate) ? solutionPathCandidate : path.resolve(workingDir, solutionPathCandidate);
+core.info(`solution path: ${solutionPath}`);
+
+const outputDirectory = path.join(process.env['RUNNER_TEMP'] || workingDir, "checker-output");
+fs.ensureDirSync(outputDirectory);
+core.info(`output directory: ${outputDirectory}`);
+
+const geo = core.getInput('geo', { required: false });
+const ruleLevelOverride = core.getInput('rule-level-override', { required: false });
+
+const logger = new ActionLogger();
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+(async () => {
+    if (!fs.existsSync(solutionPath)) {
+        throw new Error(`The solution file could not be found at ${solutionPath}`);
+    }
+
+    const pac = new PacRunner(workingDir, logger);
+    await new AuthHandler(pac).authenticate(AuthKind.CDS);
+
+    const checkArgs = ['solution', 'check', '--path', solutionPath, '--outputDirectory', outputDirectory];
+    if (geo) {
+        checkArgs.push('--geo', geo);
+    }
+    if (ruleLevelOverride) {
+        checkArgs.push('--ruleLevelOverride', ruleLevelOverride);
+    }
+    await pac.run(checkArgs);
+
+    const artifactClient = artifact.create();
+    const artifactName = 'pac-solution-check';
+    const files = fs.readdirSync(outputDirectory);
+    const options = { continueOnError: true };
+    await artifactClient.uploadArtifact(artifactName, files, outputDirectory, options);
+    core.info(`checked solution results in folder [${outputDirectory}] and uploaded as artifacts.`);
+    core.endGroup();
```

Comment:
nice to have, not blocking: grab the summary .md file and emit it to console for a quick overview directly in the action log (as a separate output group)?

Reply:
I'm not seeing any .md files produced, just the .zip containing the .sarif.
Perhaps the example Solutions I'm using are just too simple?

Reply:
can you check in the PS based impl: RunChecker.ps1
https://dev.azure.com/dynamicscrm/OneCRM/_git/PowerApps.AzDevOpsExtensions?path=%2Fsrc%2Fextension%2Ftasks%2Fchecker%2Fchecker-v0%2FRunChecker.ps1&version=GBmaster&line=122&lineEnd=123&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Reply:
emitting the summary info requires a pac CLI change first: the .SubmitCheck() call to the endpoint returns an AnalyzeResult that currently is not used in the pac CLI => out of scope for this PR, pls merge as is